src/transformStateWithClones.js

```diff
@@ -8,6 +8,36 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateCopy = { ...state };
+  const actionsResultArr = [];
+
+  function addingActionResultToArray() {
```

Comment:
you can do it without function, so, remove it
Also, would be better if your function:
- Produce No Side Effects
A pure function produces no side effects, which means that it can’t alter any external state.

- Immutability
JavaScript’s object arguments are references, which means that if a function were to mutate a property on an object or array parameter, that would mutate state that is accessible outside the function. Pure functions must not mutate external state.